### ISSUE TITLE
Feature Request: annotating rule for namespaces/excludedNamespaces

### ISSUE THREAD

**Issue by dulltz** [plexsystems/konstraint#106]
Like https://github.com/plexsystems/konstraint/issues/93, I'd like to support annotating rules that generate [`spec.match.namespaces` and `spec.match.excludedNamespaces`](https://github.com/open-policy-agent/gatekeeper/tree/1459e2f8eae495a0a34c529da794b3f1dda486d4#constraints).

## Examples

```rego
# @namespaces kube-system gatekeeper-system
package foo

import data.lib.core
...
```
↓
```yaml
apiVersion: constraints.gatekeeper.sh/v1beta1
kind: Foo
metadata:
  name: foo
spec:
  match:
    namespaces:
      - kube-system
      - gatekeeper-system
```

---

```rego
# @ excludednamespaces kube-system gatekeeper-system
package foo

import data.lib.core
...
```
↓
```yaml
apiVersion: constraints.gatekeeper.sh/v1beta1
kind: Foo
metadata:
  name: foo
spec:
  match:
    excludedNamespaces:
      - kube-system
      - gatekeeper-system
```

**Comment by jalseth** [comment]
Hi @dulltz, sorry for the delay in response. Looking back, I think I may have moved too quickly when moving forward with #93 as it somewhat goes against Konstraint's original goal of enabling the same policies to be used with Conftest and then deployed in the cluster with Gatekeeper, since these `@` tags add the matching metadata to the Gatekeeper resources, but won't have any effect when using Conftest.

I understand that some people may not use Conftest, and may instead test with gvk or similar tools so I'm hoping to find a middle ground. What are your thoughts on adding a `lib.matching` Rego lib with helpers that would accomplish the same matching that we're adding here, but using Rego rather than the Gatekeeper CRDs to match? This way, there is feature parity either way people decide to use the tool.

**Comment by dulltz** [comment]
I totally agree with @jalseth 's opinion.

Since users who use the Gatekeeper `match` field will have to manually edit the generated custom resource,
we should focus on the https://github.com/plexsystems/konstraint/issues/56 issue.

**Comment by jalseth** [comment]
In that case, a `--skip-constraints` CLI flag and a `@skip-constraint` option in the Rego header may be the best way forward. That way if users want to manually manage the `Constraint`s and set matchers, parameters, etc. they can.

**Comment by jalseth** [comment]
@dulltz The tag was added in #130 and will be included in the next release.

**Comment by jpreese** [comment]
`@skip-constraint` released in `v0.12.0`
